app/src/lib/stores/app-store.ts

```diff
@@ -1181,18 +1185,55 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.stopBackgroundFetching()
     this.stopPullRequestUpdater()
     this.stopAheadBehindUpdate()
+    if (enableBranchPruning()) {
```

Comment:
This check shouldn't be necessary as the teardown code only runs when `this.currentBranchPruner` is assigned, and that is wrapped in a `enableBranchPruning()` check in `startBackgroundPruner()`